### PR TITLE
Add GitHub Actions build workflow

### PR DESCRIPTION
**Pull request by NebelNidas**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+          cache: gradle
+
+      - name: Make Gradle wrapper executable
+        run: chmod +x ./gradlew
```
Comment (liach) [inline .github/workflows/build.yaml]: Can just update the file mode in git instead

Comment (NebelNidas) [reply .github/workflows/build.yaml]: I'm on Windows though, and my Linux VM broke a few days ago 😄 Can you fix the permissions on your system and PR the updated file to my branch? Or maybe @sfPlayer1 can do it himself, I'm not sure if he's on Linux

Comment (liach) [reply .github/workflows/build.yaml]: You can run
`git update-index --chmod +x gradlew` and commit changes.
chmod doesn't change the file in the git repo.

Comment (NebelNidas) [reply .github/workflows/build.yaml]: Nice, thanks!

**Review by liach — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+          cache: gradle
```
Comment (modmuss50) [inline .github/workflows/build.yaml]: I would advise removing Gradle caching, there are many secrity pitfalls with it.

Its not so bad here as there are no secrects, however in theory someone could pollute the builds from the main branch from a PR. I know its unlikely, but its not worth the risk just to save 30 seconds. (also making sure that Gradle can build from a clean slate is quite handy as well)

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (modmuss50) [inline .github/workflows/build.yaml]: Latest is going to change in the future, best to use `ubuntu-22.04` as its a stable target.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-artifacts
+          path: build/libs
+
+      - name: Cleanup Gradle Cache
+        run: |
+          rm -f ~/.gradle/caches/modules-2/modules-2.lock
+          rm -f ~/.gradle/caches/modules-2/gc.properties
```
Comment (modmuss50) [inline .github/workflows/build.yaml]: Remove, see above.

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          java-version: "17"
+          distribution: "temurin"
+          cache: gradle
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Build the project
+        run: ./gradlew build --stacktrace
```
Comment (modmuss50) [inline .github/workflows/build.yaml]: Minor thing:
```suggestion
        run: ./gradlew build --stacktrace --warning-mode=fail
```

Nice to keep ontop of Gradle deprecations.

**Review by modmuss50 — CHANGES_REQUESTED** [review]
I generally stay away from caching.

2 other minor nitpicks that im not too bothered by.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**